src/honeysql/helpers.clj

```diff
@@ -93,6 +93,12 @@
 (defhelper merge-right-join [m clauses]
   (update-in m [:right-join] concat clauses))
 
+(defhelper full-join [m clauses]
+  (assoc m :full-join clauses))
+
+(defhelper merge-full-join [m clauses]
+  (assoc m :merge-full-join clauses))
```

Comment:
Apologies, that's a typo that should read (update-in m [:full-join] concat clauses). I was trying to replicate the logic you had for other join methods and failed miserably.